runtime-modules/proposals/codex/src/lib.rs

```diff
@@ -372,31 +538,87 @@ decl_module! {
 
             print("Runtime upgrade proposal execution finished.");
         }
+
+        /// Update working group budget
+        #[weight = Module::<T>::get_update_working_group_budget_weight(&working_group, &balance_kind)]
+        pub fn update_working_group_budget(
+            origin,
+            working_group: WorkingGroup,
+            amount: BalanceOf<T>,
+            balance_kind: BalanceKind,
+        ) {
+            ensure_root(origin.clone())?;
+
+
+            let wg_budget = call_wg!(working_group<T>, budget);
+            let current_budget = Council::<T>::budget();
+
+            match balance_kind {
+                BalanceKind::Positive => {
+                    ensure!(amount<=current_budget, Error::<T>::InsufficientFundsForFundingRequest);
+
+                    call_wg!(working_group<T>, set_budget, origin.clone(), wg_budget.saturating_add(amount))?;
+                    Council::<T>::set_budget(origin, current_budget - amount)?;
+                },
+                BalanceKind::Negative => {
+                    ensure!(amount <= wg_budget, Error::<T>::InsufficientFundsForFundingRequest);
+
+                    call_wg!(working_group<T>, set_budget, origin.clone(), wg_budget - amount)?;
+                    Council::<T>::set_budget(origin, current_budget.saturating_add(amount))?;
```

Comment:
Since the balance is non-negative in the line before this I reduce the working group budget by `amount` and in this line I add that `amount` to the council budget. That's the behavior I understand is expected [from the handbook](https://joystream.gitbook.io/joystream-handbook/governance/proposals#update-working-group-budget).

Reply:
Sorry, I missed the call_wg part.